SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -3,19 +3,20 @@
 #include "EngineClasses/SpatialLoadBalanceEnforcer.h"
 #include "EngineClasses/SpatialVirtualWorkerTranslator.h"
 #include "Schema/AuthorityIntent.h"
+#include "Schema/Component.h"
 #include "SpatialCommonTypes.h"
 
 DEFINE_LOG_CATEGORY(LogSpatialLoadBalanceEnforcer);
 
 using namespace SpatialGDK;
 
-SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName& InWorkerId, const USpatialStaticComponentView* InStaticComponentView, const SpatialVirtualWorkerTranslator* InVirtualWorkerTranslator)
+SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName& InWorkerId, const USpatialStaticComponentView* InStaticComponentView, TSharedPtr<SpatialVirtualWorkerTranslator> InVirtualWorkerTranslator)
```

Comment:
Why is this a shared ptr?

Reply:
Previously the Translator was a TUniquePtr on the NetDriver and we passed it into the Enforcer by calling `.Get()` why seems like it defeats the point of UniquePtr to me though I'm probably misunderstanding something. Since I'm converting Translator to a TSharedPtr anyway, passing this is man sense to me? What do you suggest instead?

Reply:
We've had this conversation with @samiwh earlier. A summary would be: keeping it TUniquePtr in the NetDriver explicitly tells that the NetDriver owns the object. When the NetDriver gets destroyed - it destroys the object (Translator in this case) as well as the object's users (Enforcer in this case). So the object's users don't need to keep a SharedPtr to the object. Of course, the order of construction/destruction must be valid to avoid undefined behaviour.

Reply:
Reverted to UniquePtr

Reply:
Alex's comment is correct. My general rule of thumb is that shared_ptr is almost always wrong.